parachain/pallets/dot-app/src/benchmarking.rs

```diff
@@ -0,0 +1,97 @@
+
+benchmarks! {
+    // Benchmark `lock` extrinsic under worst case conditions:
+    // * The amount is successfully locked
+    // * The sender account is killed
+    // * The channel executes incentivization logic
+	lock {
+        let existential_deposit = T::Currency::minimum_balance();
+		let caller: T::AccountId = whitelisted_caller();
+        let lock_account = DotApp::<T>::account_id();
+        let recipient = H160::zero();
+
+        let balance = existential_deposit * 10u32.into();
+        // The amount is chosen such that balance - amount < existential_deposit
+        // so that the account is reaped
+        let amount = existential_deposit * 9u32.into() + 1u32.into();
+        
+        T::Currency::make_free_balance_be(&caller, balance);
+
+	}: _(RawOrigin::Signed(caller.clone()), ChannelId::Incentivized, recipient, amount)
+	verify {
+        assert!(!balance.is_zero() && !amount.is_zero());
+        assert_eq!(T::Currency::free_balance(&caller), Zero::zero());
+		assert_eq!(T::Currency::free_balance(&lock_account), amount);
+	}
+
+    // Benchmark `lock` extrinsic for the average case:
+    // * The amount is successfully locked
+    // * The sender remains alive
+    // * The channel executes incentivization logic
+    #[extra]
+	lock_sender_alive {
+        let existential_deposit = T::Currency::minimum_balance();
+		let caller: T::AccountId = whitelisted_caller();
+        let lock_account = DotApp::<T>::account_id();
+        let recipient = H160::zero();
+
+        let balance = existential_deposit * 10u32.into();
+        let amount = existential_deposit * 8u32.into();
+        
+        T::Currency::make_free_balance_be(&caller, balance);
+
+	}: lock(RawOrigin::Signed(caller.clone()), ChannelId::Incentivized, recipient, amount)
+	verify {
+        assert!(!balance.is_zero() && !amount.is_zero());
+        assert_eq!(T::Currency::free_balance(&caller), balance - amount);
+		assert_eq!(T::Currency::free_balance(&lock_account), amount);
+	}
+
+    // Benchmark `unlock` extrinsic under worst case conditions:
+    // * The amount is successfully unlocked
+	unlock {
+        let caller: H160 = ETH_CONTRACT_ADDRESS.into();
+        Address::put(caller);
+
+        let existential_deposit = T::Currency::minimum_balance();
+        let lock_account = DotApp::<T>::account_id();
+        let recipient: T::AccountId = account("recipient", 0, 0);
+        let recipient_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(recipient.clone());
+        let sender = H160::zero();
+        
+        let balance = existential_deposit * 10u32.into();
+        // The balance remaining after unlock will be < existential deposit.
+        // This shouldn't trigger account reaping because we're using
+        // the 'KeepAlive' flag. But just in case that ever changes...
+        let amount = existential_deposit * 9u32.into() + 1u32.into();
+        let amount_wrapped = wrap::<T>(amount, T::Decimals::get()).unwrap();
```

Comment:
Thought I'd add some more context around this "wrapping/unwrapping" business. My documentation for the DOT pallet was pretty non-existent, sorry about that.

Basically on Substrate, native DOT is represented using u128, with 10 decimal places. Our Wrapped DOT token living on Ethereum however is represented using u256 and has 18 decimal places. The 18 decimal places is a convenience to match ETH and most ERC20 tokens which also have 18 decimal places. So the task of `wrap()` is convert a native DOT to 18 decimal places with some arithmetic conversions.